test/hotspot/jtreg/compiler/loopopts/superword/MinMaxRed_Long.java

```diff
@@ -0,0 +1,155 @@
+ */
+
+package compiler.loopopts.superword;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Utils;
+
+import java.util.Arrays;
+import java.util.Random;
+import java.util.stream.LongStream;
+
+public class MinMaxRed_Long {
+
+    private static final Random random = Utils.getRandomInstance();
+
+    public static void main(String[] args) throws Exception {
+        TestFramework framework = new TestFramework();
+        framework.addFlags("-XX:+IgnoreUnrecognizedVMOptions",
+                           "-XX:LoopUnrollLimit=250",
+                           "-XX:CompileThresholdScaling=0.1");
+        framework.start();
+    }
+
+    @Run(test = {"maxReductionImplement"},
+         mode = RunMode.STANDALONE)
+    public void runMaxTest() {
+        runMaxTest(50);
+        runMaxTest(80);
+        runMaxTest(100);
+    }
+
+    private static void runMaxTest(int probability) {
+        long[] longs = new long[1024];
+        ReductionInit(longs, probability);
+        long res = 0;
+        for (int j = 0; j < 2000; j++) {
+            res = maxReductionImplement(longs, res);
+        }
+        if (res == 11 * Arrays.stream(longs).max().getAsLong()) {
+            System.out.println("Success");
+        } else {
+            throw new AssertionError("Failed");
+        }
+    }
+
+    @Run(test = {"minReductionImplement"},
+         mode = RunMode.STANDALONE)
+    public void runMinTest() {
+        runMinTest(50);
+        runMinTest(80);
+        runMinTest(100);
+    }
+
+    private static void runMinTest(int probability) {
+        long[] longs = new long[1024];
+        ReductionInit(longs, probability);
+        // Negating the values generated for controlling max branching
+        // allows same logic to be used for min tests.
+        longs = negate(longs);
+        long res = 0;
+        for (int j = 0; j < 2000; j++) {
+            res = minReductionImplement(longs, res);
+        }
+        if (res == 11 * Arrays.stream(longs).min().getAsLong()) {
+            System.out.println("Success");
+        } else {
+            throw new AssertionError("Failed");
+        }
+    }
+
+    static long[] negate(long[] nums) {
+        return LongStream.of(nums).map(l -> -l).toArray();
+    }
+
+    public static void ReductionInit(long[] longs, int probability) {
+        int aboveCount, abovePercent;
+
+        // Iterate until you find a set that matches the requirement probability
+        do {
+            long max = random.nextLong(10);
+            longs[0] = max;
+
+            aboveCount = 0;
+            for (int i = 1; i < longs.length; i++) {
+                long value;
+                if (random.nextLong(101) <= probability) {
+                    long increment = random.nextLong(10);
+                    value = max + increment;
+                    aboveCount++;
+                } else {
+                    // Decrement by at least 1
+                    long decrement = random.nextLong(10) + 1;
```

Comment:
Nit: I would call it `diffToMax`, because you are really just going to get a value below the `max`, and you are not decrementing the `max`. But up to you if you want to change it.